client/src/pages/Products/index.tsx

```diff
@@ -1,7 +1,100 @@
+import { useEffect, useState } from 'react';
+import { Table } from 'reactstrap';
+import {
+  getAllProductsAPI,
+  updateProductAPI,
+  deleteProductAPI
+} from '../../api/products';
+import { ModalEdit } from '../../components/ModalEdit';
+import { TableProductRow } from '../../components/TableProductRow';
+import { Product } from '../../interfaces/ProductInterface';
 import './style.css';
 
 const Products = () => {
-  return <div>Products</div>;
+  const [products, setProducts] = useState([] as Product[]);
+  const [{ editMode, productId }, setEditMode] = useState({
+    editMode: false,
+    productId: '-1'
+  });
+  const [updatedItem, setUpdatedItem] = useState();
+  const [deletedItem, setDeletedItem] = useState();
+
+  const startEditMode = (productId: string) => {
+    setEditMode({ editMode: !editMode, productId });
+  };
+
+  const update = (id: number, updatedProduct: Product) => {
+    updateProductAPI(id, updatedProduct)
+      .then(result => result.data)
+      .then(product => setUpdatedItem(product))
+      .catch(error => {
+        console.error('error', error);
+      });
+  };
+
+  const deleteProduct = (id: number) => {
+    deleteProductAPI(id)
+      .then(result => {
+        console.log(result);
+        return result.data;
+      })
+      .then(success => setDeletedItem(success))
+      .catch(error => {
+        console.log('delete error', error);
+      });
+  };
```

Comment:
We should define these functions when we use them, no need to pass them as props.